plugins/inputs/http/http.go

```diff
@@ -6,16 +6,24 @@ import (
 	"io"
 	"io/ioutil"
 	"net/http"
+	"net/http/cookiejar"
 	"strings"
 	"sync"
+	"time"
 
 	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
 	"github.com/influxdata/telegraf/internal"
 	httpconfig "github.com/influxdata/telegraf/plugins/common/http"
 	"github.com/influxdata/telegraf/plugins/inputs"
 	"github.com/influxdata/telegraf/plugins/parsers"
 )
 
+var (
+	defaultCookieAuthRenewal = 5 * time.Minute
+	defaultCookieAuthMethod  = http.MethodPost
```

Comment:
It's a matter of habits, but I would move this to the actual usage in `Init()`. Alternatively make this a constant.

Reply:
Having them as `vars` was a force of habit for changing them during tests.  I'll move them into `Init()`.